### PR TITLE
run all parallel builds when a serial build finishes

### PR DESCRIPTION
**Pull request by bparees**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1357786

@mfojtik ptal

### CODE REVIEW

**Comment by bparees** [comment]
[test]

File: pkg/build/controller/policy/policy_test.go [hunk 1]
```diff
@@ -120,3 +120,30 @@ func TestHandleComplete(t *testing.T) {
 		t.Errorf("build-3 should not have Status.StartTimestamp set")
 	}
 }
+
+func TestHandleCompleteParallel(t *testing.T) {
+	builds := []buildapi.Build{
+		addBuild("build-1", "sample-bc", buildapi.BuildPhaseComplete, buildapi.BuildRunPolicyParallel),
```
Comment (mfojtik) [inline pkg/build/controller/policy/policy_test.go]: shouldn't this be serial?

Comment (bparees) [inline pkg/build/controller/policy/policy_test.go]: it doesn't really matter, the point is to ensure that when a build completes, we immediately start all queued parallel builds.  the code path is the same regardless of what type of build completed.

but for sanity, i'll change it.

Comment (bparees) [inline pkg/build/controller/policy/policy_test.go]: actually i think i'm going to leave it since it's TestHandleCompleteParallel.

**Comment by bparees** [comment]
[merge]

**Pull request merged**